src/app/main/component/ubs/components/ubs-confirm-page/ubs-confirm-page.component.ts

```diff
@@ -16,47 +18,72 @@ export class UbsConfirmPageComponent implements OnInit, OnDestroy {
   responseStatus: string;
   orderResponseError = false;
   orderStatusDone: boolean;
+  isSpinner = true;
+
   private destroy$: Subject<boolean> = new Subject<boolean>();
 
   constructor(
+    public router: Router,
     private activatedRoute: ActivatedRoute,
     private snackBar: MatSnackBarComponent,
     private jwtService: JwtService,
     private ubsOrderFormService: UBSOrderFormService,
     private shareFormService: UBSOrderFormService,
-    public router: Router
+    private localStorageService: LocalStorageService,
+    private orderService: OrderService
   ) {}
 
-  toPersonalAccount() {
+  toPersonalAccount(): void {
     this.jwtService.userRole$.pipe(takeUntil(this.destroy$)).subscribe((userRole) => {
       const isAdmin = userRole === 'ROLE_ADMIN';
+      this.saveDataOnLocalStorage();
       this.router.navigate([isAdmin ? 'ubs-admin' : 'ubs-user', 'orders']);
     });
   }
 
   ngOnInit() {
     this.ubsOrderFormService.orderId.pipe(takeUntil(this.destroy$)).subscribe((oderID) => {
-      this.orderId = oderID;
-      this.orderResponseError = this.ubsOrderFormService.getOrderResponseErrorStatus();
-      this.orderStatusDone = this.ubsOrderFormService.getOrderStatus();
-      if (!this.orderResponseError && !this.orderStatusDone) {
-        this.saveDataOnLocalStorage();
-        this.activatedRoute.queryParams.subscribe((params) => {
-          this.orderId = params.order_id;
-          // Hardcoded. Need a logic from back-end to save orderId for saved unpaid order
-          this.orderId = '123';
-          this.responseStatus = params.response_status;
-          this.snackBar.openSnackBar('successConfirmSaveOrder', this.orderId);
-        });
-      } else if (!this.orderResponseError && this.orderStatusDone) {
-        this.saveDataOnLocalStorage();
+      if (oderID) {
+        this.orderId = JSON.parse(oderID).orderId;
```

Comment:
Because I get response  - {orderId: 3255, link: null} as string. 
And I render the order number for users in a modal window.
If I don do that I`ll get the entire response as a string.
So I get only orderId for my purpose.